packages/backend-openapi-utils/src/stub.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PromiseRouter from 'express-promise-router';
+import { ApiRouter } from './router';
+import { RequiredDoc } from './types';
+import {
+  ErrorRequestHandler,
+  RequestHandler,
+  NextFunction,
+  Request,
+  Response,
+  json,
+} from 'express';
+import { InputError } from '@backstage/errors';
+import { middleware as OpenApiValidator } from 'express-openapi-validator';
+
+type PropertyOverrideRequest = Request & {
+  [key: symbol]: string;
+};
+
+function validatorErrorTransformer(): ErrorRequestHandler {
+  return (error: Error, _: Request, _2: Response, next: NextFunction) => {
+    next(new InputError(error.message));
+  };
+}
+
+export function getDefaultRouterMiddleware() {
+  return [json()];
+}
+
+/**
+ * Create a new OpenAPI router with some default middleware.
+ * @param spec - Your OpenAPI spec imported as a JSON object.
+ * @param validatorOptions - `openapi-express-validator` options to override the defaults.
+ * @returns A new express router with validation middleware.
+ * @public
+ */
+export function createOpenApiRouter<T extends RequiredDoc>(
```

Comment:
Given that this is more of a lower level thing, perhaps we could give this a more verbose name and leave this for the generated name? So you'd import `createOpenApiRouter` from your `.../openapi.generated`, and we could for example call this one `createValidatedOpenApiRouter` or something like that? I'm feeling `initializeRouter` doesn't quite describe what it does, would like to have `create*` in there :p

Reply:
Makes sense, will update.